ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java

```diff
@@ -37,44 +38,60 @@
   protected long sentBytes = 0;
   protected final BlobProperties properties;
   protected final BlobType blobType;
+  protected ByteBuffer materializedBlobBuffer;
 
   private static final int UserMetadata_Size_InBytes = 4;
   protected static final int Blob_Size_InBytes = 8;
   protected static final int BlobType_Size_InBytes = 2;
-  protected static final short Put_Request_Version_V1 = 1;
-  // Version 2 added to support chunking for large objects, where the size of a chunk can be different from the size
-  // in the BlobProperties (which will be the size of the whole object).
   protected static final short Put_Request_Version_V2 = 2;
 
-  protected PutRequest(int correlationId, String clientId, BlobId blobId, BlobProperties properties,
-      ByteBuffer usermetadata, InputStream blobStream, long blobSize, BlobType blobType, short versionId) {
-    super(RequestOrResponseType.PutRequest, versionId, correlationId, clientId);
+  /**
+   * Public constructor that takes in a materialized blob stream to create a PutRequest
+   * @param correlationId the correlation id associated with the request.
+   * @param clientId the clientId associated with the request.
+   * @param blobId the {@link BlobId} of the blob that is being put as part of this request.
+   * @param properties the {@link BlobProperties} associated with the request.
+   * @param usermetadata the user metadata associated with the request.
+   * @param materializedBlobStream the materialized stream containing the blob data.
+   * @param blobSize the size of the blob data.
+   * @param blobType the type of the blob data.
+   */
+  public PutRequest(int correlationId, String clientId, BlobId blobId, BlobProperties properties,
```

Comment:
any reason to have to pass in a `ByteBufferInputStream` instead of a `ByteBuffer`?